### PR TITLE
Take advantage of simplified logging

### PR DESCRIPTION
**Pull request by knoepfel**
Follow-on to https://github.com/HEPCloud/decisionengine/pull/515.

### CODE REVIEW

**Comment by jcpunk** [comment]
Can you rebase off `HEAD`?

**Review by jcpunk — DISMISSED** [review]

**Comment by BrunoCoimbra** [comment]
Can we also use the new logging boilerplate model for Sources and Publishers as well?
e.g.: `FactoryClientManifests` and `JobClusteringPublisher`

**Comment by knoepfel** [comment]
Good catch, @BrunoCoimbra.  I'll update the PR.

**Comment by knoepfel** [comment]
@BrunoCoimbra, I adjusted the logging according to your suggestions.  I also adjusted various utility functions so that a logger is "passed in" to the function if logging is desired--this way, we associate the messages from the utility function with the module that uses that utility.

File: src/decisionengine_modules/glideinwms/resource_dist_plugins.py [hunk 1]
```diff
@@ -14,31 +8,35 @@
...
 
-def order_resources(resources):
+def order_resources(resources, logger=None):
     ordered_resources = []
     rss_foms = pd.DataFrame()
 
     for rss, column_name in _RESOURCE_FROM_COLUMN_MAP.items():
         fom_df = resources.get(rss)
-        logger.info(f"Ordering resources based on {rss}")
+        if logger:
```
Comment (jcpunk) [inline src/decisionengine_modules/glideinwms/resource_dist_plugins.py:17]: I'm wondering if `if logger is not None:` is more explicit about what is going on here.  I go back and forth on the `__bool__` methods, but I suspect we care more if a logger is passed, not if it "is true"....

Comment (knoepfel) [reply src/decisionengine_modules/glideinwms/resource_dist_plugins.py]: A fair point--changes made accordingly.

**Review by jcpunk — APPROVED** [review]

**Pull request merged**